src/pybind/mgr/dashboard/controllers/home.py

```diff
@@ -0,0 +1,111 @@
+    return config['config']['locale']
+
+
+DEFAULT_LANGUAGE = _get_default_language()
+DEFAULT_LANGUAGE_PATH = os.path.join(mgr.get_frontend_path(), DEFAULT_LANGUAGE)
+
+
+@Controller("/", secure=False)
+class HomeController(BaseController):
+    LANG_TAG_SEQ_RE = re.compile(r'\s*([^,]+)\s*,?\s*')
+    LANG_TAG_RE = re.compile(
+        r'^(?P<locale>[a-zA-Z]{1,8}(-[a-zA-Z0-9]{1,8})?)(;q=(?P<weight>[01]\.\d{0,3}))?$')
+    MAX_ACCEPTED_LANGS = 10
+
+    def _parse_accept_language(self):
+        if 'Accept-Language' in cherrypy.request.headers:
+            accept_lang_header = cherrypy.request.headers['Accept-Language']
+        else:
+            return [DEFAULT_LANGUAGE]
+
+        result = []
+        for i, m in enumerate(self.LANG_TAG_SEQ_RE.finditer(accept_lang_header)):
+            if i >= self.MAX_ACCEPTED_LANGS:
+                logger.debug("reached max accepted languages, skipping remaining")
+                break
+
+            tag_match = self.LANG_TAG_RE.match(m[1])
+            if tag_match is None:
+                raise cherrypy.HTTPError(400, "Malformed 'Accept-Language' header")
+            locale = tag_match.group('locale').lower()
+            weight = tag_match.group('weight')
+            if weight:
+                try:
+                    ratio = float(weight)
+                except ValueError:
+                    raise cherrypy.HTTPError(400, "Malformed 'Accept-Language' header")
+            else:
+                ratio = 1.0
+            result.append((locale, ratio))
+
+        result.sort(key=lambda l: l[0])
+        result.sort(key=lambda l: l[1], reverse=True)
+        logger.debug("language preference: %s", result)
+        return [l[0] for l in result]
+
+    def _language_dir(self, langs):
+        for lang in langs:
+            if lang in LANGUAGES_PATH_MAP:
+                logger.debug("found directory for language '%s'", lang)
+                cherrypy.response.headers['Content-Language'] = LANGUAGES_PATH_MAP[lang]['lang']
+                return LANGUAGES_PATH_MAP[lang]['path']
+
+        logger.debug("Languages '%s' not available, falling back to %s", langs, DEFAULT_LANGUAGE)
+        cherrypy.response.headers['Content-Language'] = DEFAULT_LANGUAGE
+        return DEFAULT_LANGUAGE_PATH
+
+    @Proxy()
+    def __call__(self, path, **params):
+        if not path:
+            path = "index.html"
+
+        if 'cd-lang' in cherrypy.request.cookie:
+            langs = [cherrypy.request.cookie['cd-lang'].value.lower()]
+            logger.debug("frontend language from cookie: %s", langs)
+        else:
+            langs = self._parse_accept_language()
+            logger.debug("frontend language from headers: %s", langs)
+
+        base_dir = self._language_dir(langs)
+        full_path = os.path.join(base_dir, path)
+        logger.debug("serving static content: %s", full_path)
+        if 'Vary' in cherrypy.response.headers:
+            cherrypy.response.headers['Vary'] = "{}, Content-Language"
```

Comment:
Sorry, I said this wrong: it's `Vary: Accept-Language`. Anyway, it seems that nobody takes too much care of observing this header :-/ In fact Google recommends [multiple paths for localization](https://support.google.com/webmasters/answer/189077) over multiple versions of the same content :-/